python-sdk/src/astro/databases/mssql.py

```diff
@@ -0,0 +1,382 @@
+            kwargs.update({"schema": self.table.metadata.database})
+        return MsSqlHook(mssql_conn_id=self.conn_id, **kwargs)
+
+    @property
+    def sql_type(self) -> str:
+        return "mssql"
+
+    def table_exists(self, table: BaseTable) -> bool:
+        """
+        Check if a table exists in the database
+        :param table: Details of the table we want to check that exists
+        """
+        inspector = sqlalchemy.inspect(self.sqlalchemy_engine)
+
+        schema = None
+        if table and table.metadata:
+            schema = table.metadata.schema
+        return bool(inspector.dialect.has_table(self.connection, table.name, schema=schema))
+
+    @staticmethod
+    def get_table_qualified_name(table: BaseTable) -> str:
+        """
+        Return the table qualified name.
+        :param table: The table we want to retrieve the qualified name for.
+        """
+        qualified_name_lists = [
+            table.metadata.database,
+            table.metadata.schema,
+            table.name,
+        ]
+        qualified_name = ".".join(name for name in qualified_name_lists if name)
+        return qualified_name
+
+    @property
+    def default_metadata(self) -> Metadata:
+        """Fill in default metadata values for table objects addressing mssql databases.
+        Currently, Schema is not being fetched from airflow connection for Mssql because, in Mssql,
+        databases and schema are different concepts
+        The MssqlHook only exposes schema
+        However, implementation-wise, it seems that if the MssqlHook receives a schema during
+        initialization, but it uses it as a database in the connection to Mssql:
+        """
+        # TODO: Change airflow MssqlHook to fetch database and schema separately
+        database = self.hook.get_connection(self.conn_id).schema
+        return Metadata(database=database, schema=self.DEFAULT_SCHEMA)  # type: ignore
+
+    def schema_exists(self, schema) -> bool:
+        """
+        Checks if a schema exists in the database
+        :param schema: DB Schema - a namespace that contains named objects like (tables, functions, etc)
+        """
+        schema_result = self.hook.run(
+            "SELECT schema_name FROM information_schema.schemata WHERE lower(schema_name) = lower(%(schema_name)s);",
+            parameters={"schema_name": schema.lower()},
+            handler=lambda x: [y[0] for y in x.fetchall()],
+        )
+        return len(schema_result) > 0
+
+    def run_sql(
+        self,
+        sql: str | ClauseElement = "",
+        parameters: dict | None = None,
+        autocommit: bool = False,
+        **kwargs,
+    ) -> CursorResult:
+        """
+        Return the results to running a SQL statement.
+        Whenever possible, this method should be implemented using Airflow Hooks,
+        since this will simplify the integration with Async operators.
+        :param sql: Contains SQL query to be run against database
```

Comment:
```suggestion
        since this will simplify the integration with Async operators.

        :param sql: Contains SQL query to be run against database
```

Reply:
fixed